scripts/run-plugin-nuts.js

```diff
@@ -0,0 +1,183 @@
+  const results = execSync(`npm show ${module} scripts`);
+  return results.includes('test:nuts');
+};
+
+const getNpmDependencies = (module) => {
+  const results = execSync(`npm show ${module} dependencies --json`);
+  return JSON.parse(results);
+};
+
+const getOrgAndProject = (module) => {
+  const results = execSync(`npm show ${module} homepage`);
+  if (Reflect.ownKeys(manualRepoOverrides).includes(module)) {
+    return manualRepoOverrides[module];
+  }
+  const repoUrl = new URL(results.toString().trim());
+  const [, org, repo] = repoUrl.pathname.split('/');
+  return { org, repo };
+};
+
+const orgSlug = (org, repo) => {
+  return `github/${org}`;
+};
+const projectSlug = (org, repo) => {
+  return `${orgSlug(org)}/${repo}`;
+};
+
+/**
+ * Triggers the just-nuts workflow for the named org/repo/branch
+ * @param sfdxVersion
+ * @param org
+ * @param repo
+ * @param branch
+ * @returns {Promise<*>}
+ */
+const triggerNutsForProject = async (sfdxVersion, org, repo, branch = 'main') => {
+  const body = {
+    branch: branch,
+    parameters: {
+      'run-auto-workflows': false,
+      'run-just-nuts': true,
+      sfdx_version: sfdxVersion,
+    },
+  };
+  return circle(`${circleciBaseUrl}project/${projectSlug(org, repo)}/pipeline`, { method: 'POST', json: body });
+};
+
+/**
+ * Triggers the nuts monitoring workflow
+ * @param jobData
+ * @param branch
+ * @returns {Promise<*>}
+ */
+const triggerNutsMonitor = async (jobData, branch = 'main') => {
+  const body = {
+    branch: branch,
+    parameters: {
+      'run-auto-workflows': false,
+      'run-just-nuts': true,
+      nuts_job_data: JSON.stringify(jobData),
+    },
+  };
+  return circle(`${circleciBaseUrl}project/${projectSlug('salesforcecli', 'sfdx-cli')}/pipeline`, {
+    method: 'POST',
+    json: body,
+  });
+};
+/**
+ * Makes a request to named URL. Ensures that auth token is applied.
+ * @param url
+ * @param options
+ * @returns {Promise<any>}
+ */
+const circle = async (url, options = {}) => {
+  const defaultOptions = {
+    method: 'GET',
+    headers: { 'Circle-Token': process.env.CIRCLECI_API_TOKEN, responseType: 'json', resolveBodyOnly: true },
+  };
+  options = { ...defaultOptions, ...options };
+  const response = await (async () => {
+    try {
+      return await got(url, options);
+    } catch (error) {
+      console.log(error);
+      throw new Error(error && error.response?.body ? error.response.body : error);
+    }
+  })();
+  return JSON.parse(response.body);
+};
+
+/**
+ * Examines a lit of plugins to determine which have NUT tests
+ * @param plugins
+ * @returns {*}
+ */
+const qualifyPluginsWithNonUnitTests = (plugins) => {
+  return (
+    plugins
+      // find plugins that may have NUT tests
+      .filter((plugin) => plugin === 'salesforcedx' || plugin.startsWith('@salesforce'))
+      // handle sub-aggregate plugins by fetching that plugins dependencies
+      .map((plugin) => {
+        if (plugin === 'salesforcedx') {
```

Comment:
You could probably also make this generic and even recursive if you wanted. If `npm view <pkg> oclif.plugins` then also check those. I think it is fine to also keep it simple for now and make it more generic later if needed.